api/app/services/dashboard_service.py

```
@@ -340,6 +431,137 @@ async def _calculate_negative_feedback_trend(self) -> float:
+
+        elif period == "30d":
+            # Current: last 30 days
+            current_start = now - timedelta(days=30)
+            current_end = now
+            # Previous: 30-60 days ago
+            previous_start = now - timedelta(days=60)
+            previous_end = now - timedelta(days=30)
+
+        elif period == "custom":
+            if not start_date or not end_date:
+                raise ValueError("start_date and end_date required for custom period")
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)